plugins/genesys/fu-genesys-usbhub-device.c

```diff
@@ -816,7 +859,9 @@ fu_genesys_usbhub_device_setup(FuDevice *device, GError **error)
 				   sizeof(FuGenesysStaticToolString));
 	}
 
-	if (memcmp(self->static_ts.mask_project_ic_type, "3523", 4) == 0) {
+	if (memcmp(self->static_ts.mask_project_ic_type, "3521", 4) == 0) {
+		self->chip.model = ISP_MODEL_HUB_GL3521;
```

Comment:
oh, this is strange. `7364` -> GL3521-2x and `7411` -> `GL3522-XX`. The two devices I have says `352122` and `7411` are they correct?

Reply:
btw, are they very different? If they are quite similar and since they use the `ic_type`, it makes sense to have them both under `ISP_MODEL_HUB_GL3521`.

Reply:
The series has GL3520-21 , GL3521-21 , GL3522-21 , GL3520-22 , GL3521-22 , GL3522-22.

I tested below. I can check it in-house.
7364 -> 352021
7411 -> 352122

Reply:
or even better the following, if `code` is a decimal incremental number:

``` diff
        }

        if (memcmp(self->static_ts.mask_project_ic_type, "3521", 4) == 0) {
-               self->chip.model = ISP_MODEL_HUB_GL3521;
+               gchar tmp[5] = {0};
+               memcpy(tmp, &self->static_ts.mask_project_code, 4);
+               code = fu_common_strtoull(tmp);
+
+               if (code < 7411)
+                       self->chip.model = ISP_MODEL_HUB_GL3521;
+               else
+                       self->chip.model = ISP_MODEL_HUB_GL3522;
        } else if (memcmp(self->static_ts.mask_project_ic_type, "3523", 4) == 0) {
                self->chip.model = ISP_MODEL_HUB_GL3523;
        } else if (memcmp(self->static_ts.mask_project_ic_type, "3590", 4) == 0) {
```

Reply:
However, I suggest to stick to the original change; and make the changes for supporting GL3520.

Reply:
From the result I tested, maybe 7364 is 352X-21, 7411 is 352X-22. I should check the rule inside. But the model is the same one for me.

>7364 -> 352021
7411 -> 352122

Reply:
Ok. And what about the GL3520.

Let's stick to what we have now.

Reply:
> Ok. And what about the GL3520.
> 
> Let's stick to what we have now.

GL3520-10 is a legacy product and EOL.